demos/object_detection_demo/cpp/proposal.cpp

```diff
@@ -0,0 +1,113 @@
+DEFINE_uint32(nthreads, 0, "Number of threads to use for inference on the CPU (including HETERO and MULTI cases)");
+DEFINE_string(nstreams, "",
+    "Number of streams to use for inference on the CPU, GPU or MYRIAD devices "
+    "(for HETERO and MULTI device cases use format <dev1>:<nstreams1>,<dev2>:<nstreams2> or just <nstreams>). "
+    "Default value is determined automatically for a device.Please note that although the automatic selection "
+    "usually provides a reasonable performance, it still may be non - optimal for some cases, especially for "
+    "very small networks. See sample's README for more details. "
+    "Also, using nstreams>1 is inherently throughput-oriented option, "
+    "while for the best-latency estimations the number of streams should be set to 1");
+DEFINE_bool(show, false, "-noshow: don't show output");
+DEFINE_string(u, "", "List of monitors to show initially");  // TODO: more clear wording
+DEFINE_bool(reverse_input_channels, false, "Switch the input channels order from BGR to RGB");
+DEFINE_string(mean_values, "", "Normalize input by subtracting the mean values per channel. Example: '255.0 255.0 255.0'");
+DEFINE_string(scale_values, "",
+    "Divide input by scale values per channel. Division is applied "
+    "after mean values subtraction. Example: '255.0 255.0 255.0'");
+
+void parse(int argc, char *argv[]) {
+    gflags::ParseCommandLineFlags(&argc, &argv, false);
+    slog::info << ov::get_openvino_version() << slog::endl;
+    if (FLAGS_h || argc == 1) {
+        exit(0);
+    }
+}
+
+unique_ptr<Processor> create_processor(const string& aarchitecture_type, const string& weights,
+        double confidence_threshold, const vector<string>& labels,
+        bool reverse_nput_channels, const string &mean_values, const string &scale_values) {
+    return unique_ptr<Processor>(new SSD(weights, float(confidence_threshold), true, labels));
+}
+
+cv::Mat render_detection_data(DetectionPredictions& predictions, const ColorPalette& palette, OutputTransform& output_transform) {
+    return cv::Mat{};
+}
+}  // namespace
+
+int main(int argc, char *argv[]) {
+    set_terminate(catcher);
+    parse(argc, argv);
+    shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop);
+    ColorPalette palette;
+    OutputTransform output_transform;
+    LazyVideoWriter video_writer;
+    Presenter presenter(FLAGS_u);
+    PerformanceMetrics render_metrics;
+    PerformanceMetrics metrics;
+    vector<string> labels;
+    if (!FLAGS_labels.empty()) {
+        labels = DetectionModel::loadLabels(FLAGS_labels);
+    }
+    SingleProcessorInferer inferer(
+        create_processor(FLAGS_at, FLAGS_m, FLAGS_t, labels,
+            FLAGS_reverse_input_channels, FLAGS_mean_values, FLAGS_scale_values),
+        ConfigFactory::getUserConfig(FLAGS_d, "", "", FLAGS_nireq, FLAGS_nstreams, FLAGS_nthreads));
+
+    for (const shared_ptr<State>& state : inferer) {
```

Comment:
I think this can be resolved as it duplicates the problem idea from the second paragraph of https://github.com/openvinotoolkit/open_model_zoo/pull/3110#discussion_r792443089 Which I'm going to ignore because of
> I suppose after a discussion I need to consider only the first paragraph.